vidyut-prakriya/src/samjna.rs

```diff
@@ -409,9 +412,9 @@ fn try_run_for_dhatu_pratyaya(p: &mut Prakriya, i: usize) -> Option<()> {
     })?;
 
     if pratyaya.is_pratyaya() {
-        if pratyaya.has_lakara(Lit) {
+        if pratyaya.has_lakara(Lit) && !pratyaya.is_ardhadhatuka() {
             p.add_tag_at("3.4.115", i, T::Ardhadhatuka);
-        } else if pratyaya.has_lakara(AshirLin) {
+        } else if pratyaya.has_lakara(AshirLin) && !pratyaya.is_ardhadhatuka() {
```

Comment:
lit and ashirlin are always ardhadhatuka -- or is the issue that we've run these rules already?